### PR TITLE
[Chart Enhancement] Fix Eggnog Erect ending notes + small adjustments

### PR DESCRIPTION
**Pull request by afreetoplaynoob**
The ending of Eggnog Erect was using the wrong BPM (80, instead of 70, half of the song's main speed) and the four ending notes were off-sync, so I fixed those things and made some other changes to the chart.

Fixes this in both Erect and Nightmare difficulties.

Screenshot 2024-11-20 23 17 07

### CODE REVIEW

File: preload/data/songs/eggnog/eggnog-metadata-erect.json [hunk 1]
```diff
@@ -2,33 +2,72 @@
   "version": "2.2.4",
   "songName": "Eggnog Erect",
   "artist": "Saruky",
```
Comment (Hundrec) [inline preload/data/songs/eggnog/eggnog-metadata-erect.json:4]:
```suggestion
  "artist": "Saruky",
  "charter": "fabs + SpazKid",
```

Heads up, exporting charts removes the charter from the Metadata!

**Review by Hundrec — CHANGES_REQUESTED** [review]

File: preload/data/songs/eggnog/eggnog-metadata-erect.json [hunk 2]
```diff
@@ -2,33 +2,72 @@
...
+    "vocals": {    },
+    "altVocals": {    }
+  },
   "playData": {
-    "difficulties": ["erect", "nightmare"],
+    "songVariations": [    ],
+    "difficulties": [
+      "erect",
+      "nightmare",
+      "hard"
```
Comment (Hundrec) [inline preload/data/songs/eggnog/eggnog-metadata-erect.json:18]:
```suggestion

```
Exporting charts from erect variations also creates an excess Hard difficulty with no notes.

File: preload/data/songs/eggnog/eggnog-chart-erect.json [hunk 3]
```diff
@@ -1,2086 +1,11166 @@
...
-      }
+	"duration": 46,
+	"ease": "quadInOut",
+	"zoom": 0.82,
+	"mode": "stage"
+}
     }
   ],
   "notes": {
+    "hard": [    ],
```
Comment (Hundrec) [inline preload/data/songs/eggnog/eggnog-chart-erect.json:940]:
```suggestion

```
Removing this excess Hard difficulty should make it good to go!

**Review by Hundrec — CHANGES_REQUESTED** [review]

**Comment by Hundrec** [comment]
No problem!
I've been curious: What is the full list of changes you made to the chart?
Ideally, we would be able to go in and manually edit the notes and events without re-exporting the song and making comparison difficult. (See line changes for #96 as an example)

**Comment by afreetoplaynoob** [comment]
Excluding the aformentioned BPM change and ending notes:

## Section 11 - Parents (Christmas)
Added a 32nd down note to accurately chart the grace.

Screenshot 2024-11-21 10 53 32

## Section 15 - Parents (Christmas)
Added a 32nd right note to accurately chart the grace.

Screenshot 2024-11-21 10 54 27

## Section 47 - Parents (Christmas)
Added an up note for the small note Daddy Dearest sings here.

Screenshot 2024-11-21 10 55 21

## Section 56 - Boyfriend (Christmas)
Removed the down note grace here and moved the right note a bit earlier.

Screenshot 2024-11-21 11 01 02

## Section 30 - Parents (Christmas) [ERECT Exclusive]
Moved the three notes immediately after the event a bit lower to sync with the singing.

Screenshot 2024-11-21 11 04 18

## Note
The section 56 change isn't in the PR itself yet, as I only figured out the proper timing for the note when gathering screenshots for this comment.

**Comment by Hundrec** [comment]
Yeah, the grid is super important!
I'll clarify that it looks like the new notes you've placed align in 32nds with the regular variation's 150bpm grid, affecting the accuracy of the placements. Please revise these placements on the correct grid and update me with the changes!

**Comment by afreetoplaynoob** [comment]
Alright, here's the changes again, now in their _proper_ positions and sections, with the changed notes highlighted and already manually edited into the original file:

## Section 10 - Parents

Screenshot 2024-11-21 20 26 23

## Section 14 - Parents

Screenshot 2024-11-21 20 26 40

## Section 44 - Parents

Screenshot 2024-11-21 20 26 57

## Section 52 - Boyfriend

Screenshot 2024-11-21 20 27 06

## Section 71 - Both

Here's how it looks like in 70 BPM:
Screenshot 2024-11-21 20 27 33
And how it looks like in 140 BPM, for easier viewing:
Screenshot 2024-11-21 20 29 12

## Section 28 - Boyfriend [ERECT Only]

Screenshot 2024-11-21 20 27 54

**Comment by Hundrec** [comment]
Thanks so much! I'll look into these changes when I have the time and create a branch in my fork that manually edits the json files without re-exporting the chart.

That way we can use the GitHub diffs for comparison and save some disk space for the game!

**Comment by Hundrec** [comment]
Oh whoops, didn't notice you had noted that you already made those manual edits.
Can you commit those changes to this PR so I can examine them?

**Comment by Hundrec** [comment]
*Also* just noticed that you already made a commit for those manual edits.
I wonder why GitHub is displaying the diffs as if they're replacing entire chunks instead of just the parts you changed...

**Comment by afreetoplaynoob** [comment]
Pretty sure it's from all the changes and addendums we've made to the file; this PR started with an exported chart file and then went back to the original file format, so it's seeing it as the diffs to the exported format.

**Comment by Hundrec** [comment]
Got it, then I will manually copy over your files to my fork and see if the diffs are small there

**Comment by Hundrec** [comment]
Just checked out your changes and they're great!
If you can squash this PR into one commit, it should look like [this branch](https://github.com/Hundrec/funkin.assets/commit/bc31b8983b222d9a7763d062c6fc1c1192fd337e) and be easily comparable.

**Comment by afreetoplaynoob** [comment]
Looks like I miscalculated there, the diffs are still messed up...
At this point I might as well delete the repo and close this PR so I can make a new one with good diffs.

**Pull request closed without merging**